R/sample_data.R

```diff
@@ -63,15 +63,28 @@ read_sample_data <- function(p, results_dir, tx2gene = NULL) {
       dragen_fusions <- NULL
     }
   }
-  salmon <- salmon_counts(salmon, tx2gene = tx2gene)
   dragen_fusions <- dragen_fusions_read(dragen_fusions)
   dragen_mapping_metrics <- dragen_mapping_metrics_read(dragen_mapping_metrics)
+  #---- Kallisto ----#
+  kallisto <- p[["kallisto"]]
+
+  # check which quant input is provided
+  if (!is.null(salmon)) {
+    counts <- salmon_counts(salmon, tx2gene = tx2gene)
+  } else if (!is.null(kallisto)) {
+    counts <- kallisto_counts(kallisto, tx2gene = tx2gene)
+  } else if (!is.null(kallisto) && !is.null(salmon)) {
+    counts <- NULL
+  } else {
+    counts <- NULL
+  }
```

Comment:
Looking at the logic here, let's say we have the booleans A and B. This is what's happening here:

```
  if (A) { # if salmon is specified
    ...
  } else if (B) { # so salmon is not specified but kallisto is
    ...
  } else if (A && B) { # so salmon is not specified, and kallisto is not specified, but you're checking if both of them are specified
    ...
  } else {
    ...
  }
```